src/scripts/main.js

```diff
@@ -1,3 +1,83 @@
 'use strict';
 
 // write code here
+const appendRowButtonRef = document.querySelector('.append-row');
+const removeRowButtonRef = document.querySelector('.remove-row');
+const appendColumnButtonRef = document.querySelector('.append-column');
+const removeColumnButtonRef = document.querySelector('.remove-column');
+const tableRef = document.querySelector('.field');
+const tbody = tableRef.children[0];
+
+removeRowButtonRef.addEventListener('click', () => {
+  const isAppendButtonDisabled = appendRowButtonRef.disabled;
+  const lastRow = tbody.lastElementChild;
+  const numberOfRows = tbody.children.length - 1;
+
+  if (numberOfRows <= 2) {
+    removeRowButtonRef.disabled = true;
+  }
+
+  if (isAppendButtonDisabled) {
+    appendRowButtonRef.disabled = !isAppendButtonDisabled;
+  }
+
+  lastRow.remove();
+});
+
+appendRowButtonRef.addEventListener('click', () => {
+  const isRemoveButtonDisabled = removeRowButtonRef.disabled;
+  const newTableRow = document.createElement('tr');
+  const numberOfRows = tbody.children.length + 1;
+
+  newTableRow.innerHTML = tbody.children[0].innerHTML;
+
+  if (numberOfRows >= 10) {
+    appendRowButtonRef.disabled = true;
+  }
+
+  if (isRemoveButtonDisabled) {
+    removeRowButtonRef.disabled = !isRemoveButtonDisabled;
+  }
+
+  tbody.append(newTableRow);
+});
+
+appendColumnButtonRef.addEventListener('click', () => {
+  const isRemoveButtonDisabled = removeColumnButtonRef.disabled;
+  const arrayOfTableRows = [...tbody.children];
+  const numberOfCols = arrayOfTableRows[0].children.length + 1;
+
+  if (numberOfCols >= 10) {
+    appendColumnButtonRef.disabled = true;
+  }
```

Comment:
Recommend to make global function from this, like: 

`const disable = (element, condition, value) => condtion && element.disabled = value;`

And use instead of these `if`